python-backend/app/api/mines/mine/resources/mine_type.py

```diff
@@ -1,59 +1,59 @@
 import uuid
-
 from sqlalchemy import exc
 from flask_restplus import Resource, reqparse
+from werkzeug.exceptions import BadRequest, NotFound
+
 from app.extensions import api
 from ....utils.access_decorators import requires_role_mine_create
-from ....utils.resources_mixins import UserMixin, ErrorMixin
+from ....utils.resources_mixins import UserMixin
 from ..models.mine_type import MineType
+from app.api.mines.mine_api_models import MINE_TYPE_MODEL
 
 
-class MineTypeResource(Resource, UserMixin, ErrorMixin):
+class MineTypeListResource(Resource, UserMixin):
     parser = reqparse.RequestParser(trim=True)
     parser.add_argument(
         'mine_guid',
         type=str,
-        help='Unique identifier for the mine with which to associate this mine type.')
-    parser.add_argument('mine_tenure_type_code', type=str, help='Mine tenure type identifier.')
+        help='Unique identifier for the mine with which to associate this mine type.',
+        location='json',
+        required=True)
+    parser.add_argument(
+        'mine_tenure_type_code',
+        type=str,
+        help='Mine tenure type identifier.',
+        location='json',
+        required=True)
 
     @api.expect(parser)
+    @api.marshal_with(MINE_TYPE_MODEL, code=201)
+    @api.doc(description='Creates a mine type and associates it with a Mine.')
     @requires_role_mine_create
     def post(self):
         data = self.parser.parse_args()
 
         mine_guid = data['mine_guid']
         mine_tenure_type_code = data['mine_tenure_type_code']
 
-        if not mine_guid:
-            self.raise_error(400, 'Error: Missing mine_guid.')
+        mine_type = MineType.create_mine_type(
+            mine_guid, mine_tenure_type_code, add_to_session=False)
+        mine_type.save()
 
-        if not mine_tenure_type_code:
-            self.raise_error(400, 'Error: Missing mine_tenure_type_code.')
+        return mine_type
 
-        try:
-            mine_type = MineType.create_mine_type(
-                mine_guid, mine_tenure_type_code, add_to_session=False)
-            mine_type.save()
-        except exc.IntegrityError as e:
-            self.raise_error(400, 'Error: Unable to create mine_type.')
 
-        return mine_type.json()
-
-    @api.expect(parser)
+class MineTypeResource(Resource, UserMixin):
+    @api.doc(description='Deletes the mine type provided.')
```

Comment:
code=204 for swagger docs ?

Reply:
I have added in a api.response() so there is a code displaying now.